Framework/PythonInterface/plugins/algorithms/SaveReflections.py

```diff
@@ -0,0 +1,259 @@
+import numpy as np
+from mantid.api import AlgorithmFactory, FileProperty, FileAction, PythonAlgorithm, ITableWorkspaceProperty
+from mantid.kernel import StringListValidator, Direction
+from mantid.simpleapi import SaveHKL
+
+# List of file format names supported by this algorithm
+SUPPORTED_FORMATS = ["Fullprof", "GSAS", "Jana", "SHELX"]
+NUM_PEAKSWS_COLUMNS = 17
+
+
+def has_modulated_indexing(workspace):
+    """Check if this workspace has more than 3 indicies
+
+    :params: workspace :: the workspace to check
+    :returns: True if the workspace > 3 indicies else False
+    """
+    return workspace.columnCount() > NUM_PEAKSWS_COLUMNS
+
+
+def num_additional_indicies(workspace):
+    """Check if this workspace has more than 3 indicies
+
+    :params: workspace :: the workspace count indicies in
+    :returns: the number of additional indicies present in the workspace
+    """
+    return workspace.columnCount() - NUM_PEAKSWS_COLUMNS
+
+
+def get_additional_index_names(workspace):
+    """Get the names of the additional indicies to export
+
+    :params: workspace :: the workspace to get column names from
+    :returns: the names of any additional columns in the workspace
+    """
+    return ["m{}".format(i+1) for i in range(num_additional_indicies(workspace))]
+
+
+class SaveReflections(PythonAlgorithm):
+
+    def category(self):
+        return "DataHandling"
+
+    def summary(self):
+        return "Saves single crystal reflections to a variety of formats"
+
+    def PyInit(self):
+        """Initilize the algorithms properties"""
+
+        self.declareProperty(FileProperty("Filename", "",
```

Comment:
I would move `Filename` to after `InputWorkspace` - this is more consistent with other `Save*` algorithms.

Reply:
Agreed.